security-claims/src/violation.rs

```diff
@@ -1,11 +1,35 @@
+use crate::{Claim, ClaimCipher, ClaimType};
 use itertools::Itertools;
-use security_claims::{Claim, ClaimCipher, ClaimType};
-
-use crate::agent::{AgentName, TLSVersion};
-
-type ClaimMessage = (AgentName, Claim);
+use std::any::Any;
+use std::fmt::Debug;
+use std::hash::Hash;
+
+// Will be instantiated with (AgentName,Claim)
+pub type ClaimMessage<AgentName>
+where
+    AgentName: Copy + Clone + Debug + Eq + PartialEq + Hash,
```

Comment:
```suggestion
    A: AgentName,
```

We should create here a trait `AgentName`:

```rust
trait AgentName: Copy + Clone + Debug + Eq + PartialEq + Hash {}
```